plugins/TimingTriggerCandidateMaker.cpp

```diff
@@ -37,48 +38,87 @@ TimingTriggerCandidateMaker::TimingTriggerCandidateMaker(const std::string& name
-    candidate.time_start = data.timestamp - m_hsi_pt_before;
-    candidate.time_end = data.timestamp + m_hsi_pt_after;
-  } else {
-    if (m_detid_offsets_map.count(data.signal_map)) {
-      // clang-format off
-      candidate.time_start = data.timestamp - m_detid_offsets_map[data.signal_map].first;  // time_start
-      candidate.time_end   = data.timestamp + m_detid_offsets_map[data.signal_map].second; // time_end,
-      // clang-format on    
-    } else {
+
+  std::vector<int> triggered_bits = GetTriggeredBits(data.signal_map);
+  for (int bit_index : triggered_bits) {
+    uint32_t signal = 1 << bit_index;
+
+    if (!m_hsisignal_map.count(signal)) {
       throw dunedaq::trigger::SignalTypeError(ERS_HERE, get_name(), data.signal_map);
     }
-  }
-  candidate.time_candidate = data.timestamp;
-  // throw away bits 31-16 of header, that's OK for now
-  candidate.detid = { static_cast<triggeralgs::detid_t>(data.signal_map) }; // NOLINT(build/unsigned)
-  candidate.type = triggeralgs::TriggerCandidate::Type::kTiming;
 
-  candidate.algorithm = triggeralgs::TriggerCandidate::Algorithm::kHSIEventToTriggerCandidate;
-  candidate.inputs = {};
+    triggeralgs::TriggerCandidate candidate;
+    candidate.time_start = data.timestamp - m_hsisignal_map[data.signal_map].time_before;  // time_start
+    candidate.time_end   = data.timestamp + m_hsisignal_map[data.signal_map].time_after; // time_end,
+    candidate.time_candidate = data.timestamp;
+    // throw away bits 31-16 of header, that's OK for now
+    candidate.detid = { static_cast<triggeralgs::detid_t>(signal) }; // NOLINT(build/unsigned)
+
+    candidate.type = m_hsisignal_map[signal].type; // type,
+
+    candidate.algorithm = triggeralgs::TriggerCandidate::Algorithm::kHSIEventToTriggerCandidate;
+    candidate.inputs = {};
+    candidates.push_back(candidate);
+  }
 
-  return candidate;
+  return candidates;
 }
 
 
 void
 TimingTriggerCandidateMaker::do_conf(const nlohmann::json& config)
 {
   auto params = config.get<dunedaq::trigger::timingtriggercandidatemaker::Conf>();
-  m_detid_offsets_map[params.s0.signal_type] = { params.s0.time_before, params.s0.time_after };
-  m_detid_offsets_map[params.s1.signal_type] = { params.s1.time_before, params.s1.time_after };
-  m_detid_offsets_map[params.s2.signal_type] = { params.s2.time_before, params.s2.time_after };
-  m_hsi_passthrough = params.hsi_trigger_type_passthrough;
-  m_hsi_pt_before = params.s0.time_before;
-  m_hsi_pt_after = params.s0.time_after;
+
+  // Fill the internal hsi signal map. The map contains signal ID, TC type to
+  // create, and the readout time window.
+  for (auto hsi_input : params.hsi_configs) {
+    triggeralgs::TriggerCandidate::Type type;
+    type = static_cast<triggeralgs::TriggerCandidate::Type>(
+        dunedaq::trgdataformats::string_to_fragment_type_value(hsi_input.tc_type_name));
+    if (type == triggeralgs::TriggerCandidate::Type::kUnknown) {
+      throw ConfigurationProblem(ERS_HERE,
```

Comment:
ditto regarding the issue name

Reply:
I just wanted a very generic but costomizable "there's a config issue" message here, should I make a separate issue for this?

Reply:
I'm not sure whether there are rules for how the issues are to be defined. I *think* from looking at existing ones that the idea is to be as specific as possible - including passing information about the variables involved. I'd be happy with a global config issue, as long as it is contained for the plugin and has a reference to it.

Reply:
Changed, but as before, I don't like that change.